limiter.go

```diff
@@ -72,12 +72,35 @@ func (l *RateLimiter) OnLimit(w http.ResponseWriter, r *http.Request, key string
 	currentWindow := time.Now().UTC().Truncate(l.windowLength)
 	ctx := r.Context()
 
-	limit := l.requestLimit
-	if val := getRequestLimit(ctx); val > 0 {
-		limit = val
+	setHeader(w, l.headers.Reset, fmt.Sprintf("%d", currentWindow.Add(l.windowLength).Unix()))
+
+	limit, ok := getRequestLimit(ctx)
+	if !ok {
+		limit = l.requestLimit
+	}
+	// If the limit is set to 0, we are always over limit
+	if limit == 0 {
+		return true
+	}
+	// If the limit is set to -1, we are never over limit
+	if limit == _NoLimit {
+		return false
 	}
+
+	increment, ok := getIncrement(r.Context())
+	if !ok {
+		increment = 1
+	}
+	// If the increment is 0, we are always on limit
+	if increment == 0 {
+		return false
```

Comment:
we should set `X-RateLimit-Increment: 0`

Reply:
It's now done here: https://github.com/go-chi/httprate/pull/46/files#diff-c3f1fd6fa2457438e1cd957eea2ba705f9825a4b6e3ad3fa55e3e03be05dea05R96-R98